### ISSUE TITLE
where can I get the val loss?

### ISSUE THREAD

**Issue by FrankFang0813** [gathierry/FastFlow#20]
Hi, first, thank you for your code!!!

I want to use val loss to choose the best ckpt, I think "loss_meter.avg" is training loss.
So where can I get the val loss?
thank you!!

**Comment by gathierry** [comment]
You can get the loss value from `ret` here with `ret["loss"]` https://github.com/gathierry/FastFlow/blob/37c99ae33b8db0e5cb534fd0d197b0bc44817a03/main.py#L96

Then build another `loss_meter` in `eval_once` function to aggregate the losses.
